### PR TITLE
[FIX] azure_oauth: configuration mail server Outlook.

### PR DESCRIPTION
**Pull request by Abridbus**
Results of seeing a lot of wrong and missing configuration on consultancy project for Outlook accounts.

The Multi outgoing mail server is harder to maintain than having a unique mail server with an Odoo db correctly configured.

Insisting on the mail.default.from (ICP) and the From Filtering parameters that NEEDS TO BE set up.

task-3512379

### CODE REVIEW

**Comment by StraubCreative** [comment]
@tiku-odoo can you verify the content and then tag @samueljlieber for technical review, please?

**Review by jqu-odoo — COMMENTED** [review]
Here is my own summary of the issue and perhaps the basis for a different wording.

**Configuration with a single outgoing mail server**

You use a generic email address, e.g. notifications@example.com or odoo@example.com, connected as an outgoing mail server.

This address must be set as from_filter on the server.

This address must also match the mail.default.from@mail.catchall.domain key combination in the system parameters.

Configuration sample :

- Outgoing mail server login = notifications@example.com
- Outgoing mail server From filter = notifications@example.com
- mail.catchall.domain in system parameters = example.com
- mail.default.from in system parameters = notifications

**Multiple user configuration**

In addition to the generic server, you can use an outgoing mail server for each user. The email address of each user must be set as from_filter on the server.

Sample configuration :

- Outgoing mail server 1 login = notifications@example.com
- Outgoing mail server 1 From filter = notifications@example.com
- Outgoing mail server 2 login = john@example.com
- Outgoing mail server 2 From filter = john@example.com
- Outgoing mail server 3 login = jane@example.com
- Outgoing mail server 3 From filter = jane@example.com
- mail.catchall.domain in system parameters = example.com
- mail.default.from in system parameters = notifications

Bottom line: we see a lot of misconfigurations because mail.default.from is edited to use a full email address, which does not belong to mail.catchall.domain. mail.catchall.domain should be edited to match the From filter.

**Comment by tiku-odoo** [comment]
@Abridbus @jqu-odoo 

I've updated this PR to include the info you mentioned in your previous comments/edits. Can you review and stamp your approval before we move this over for a technical (RST) Markdown review? 

Thanks in advance for your help on this! 

:+1: 

 CC: @StraubCreative

**Comment by tiku-odoo** [comment]
@jqu-odoo -- Hi Jonathan, when you have a moment, can you review this doc for technical accuracy? Thanks in advance for your help on this. Great suggestions so far! 

Thanks, Tim :+1:

File: content/administration/maintain/azure_oauth.rst [hunk 1]
```diff
@@ -188,24 +188,102 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
-Each user should have a separate server set up. The :guilabel:`from-filter` should be set so that
-only the user's email is sent from that server. In other words, only a user with an email address
-that matches the set :guilabel:`from-filter` is able to use this server.
+In addition to a generic email server, individual email servers can be set up for users in a
+database. These email addresses must be set as the :guilabel:`FROM Filtering` on each individual
+server for this configuration to work.
+
+This configuration is the more difficult of the two Microsoft Azure configurations, in that it
+requires all users configured with email servers to have access rights to settings to establish a
+connection to the email server.
```
[the remark this answers is not in the document]
Comment (tiku-odoo) [reply content/administration/maintain/azure_oauth.rst:239]: I have clarified this sentence a bit. I am leaving the work-around out for now, since this isn't standard anymore. 

Let me know what you think. 

Thanks,
Tim

File: content/administration/maintain/azure_oauth.rst [hunk 2]
```diff
@@ -188,24 +188,102 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+
+Configuration with a single outgoing mail server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Configuring a single outgoing server is the simplest configuration available for Microsoft Azure
+and it doesn't require extensive access rights for the users in the database.
+
+A generic email address would be used to send emails for all users within the database. For example
+it could be structured with a `notifications` alias (`notifications@example.com`) or `info` alias
+(info@example.com). This address must be set as the :guilabel:`FROM Filtering` on the server. This
```
[the remark this answers is not in the document]
Comment (tiku-odoo) [reply content/administration/maintain/azure_oauth.rst:201]: Noted, I've corrected this.

File: content/administration/maintain/azure_oauth.rst [hunk 3]
```diff
@@ -188,24 +188,102 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+.. example::
+   Single outgoing mail server configuration:
+
+   - Outgoing mail server login = `notifications@example.com`
+   - Outgoing mail server From filter = `notifications@example.com`
+   - `mail.catchall.domain` in system parameters = `example.com`
+   - `mail.default.from` in system parameters = `notifications`
+
 Multiple user configuration
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
[the remark this answers is not in the document]
Comment (tiku-odoo) [reply content/administration/maintain/azure_oauth.rst:231]: I've added this in, thanks for the suggestion.

File: content/administration/maintain/azure_oauth.rst [hunk 4]
```diff
@@ -188,24 +188,102 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+   - Outgoing mail server #2 From filter = `john@example.com`
+   - Outgoing mail server #3 login = `jane@example.com`
+   - Outgoing mail server #3 From filter = `jane@example.com`
+   - `mail.catchall.domain` in system parameters = `example.com`
+   - `mail.default.from` in system parameters = `notifications`
+
+.. warning::
+   Misconfiguration can occur when the `mail.default.from` system parameter is set to use a full
+   email address that doesn't belong to the `mail.catchall.domain` system parameter. The
+   `mail.catchall.domain` system parameter should be edited to match the From Filter.
```
[the remark this answers is not in the document]
Comment (tiku-odoo) [reply content/administration/maintain/azure_oauth.rst:285]: How would you configure this?

[the remark this answers is not in the document]
Comment (Abridbus) [reply content/administration/maintain/azure_oauth.rst:285]: This is not a reply, it's an explanation of no configuration to do.

As long as we don't do a FAQ, I would not mention it.

[the remark this answers is not in the document]
Comment (tiku-odoo) [reply content/administration/maintain/azure_oauth.rst:285]: Removed this warning block per @Abridbus

:+1:

**Comment by tiku-odoo** [comment]
@samueljlieber This doc is ready for your technical review.

File: content/administration/maintain/azure_oauth.rst [hunk 5]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+   .. seealso::
+      For more information on setting the mass-mailing email server visit :ref:`this documentation
+      <email_communication/mass_mails>`.
+
+.. example::
+   Multiple user outgoing mail server configuration:
+
+   - Notifications mailbox
+      - Outgoing mail server #1 **username** (login) = `notifications@example.com`
+      - Outgoing mail server #1 :guilabel:`FROM Filtering` = `notifications@example.com`
```
Comment (Abridbus) [inline content/administration/maintain/azure_oauth.rst:275]: Can you put this Notifications mailbox AFTER the 2 User ?
- Visually speaking it makes more sense (this is a fallback)
- it matches the explanation given previously (first configure 1 OMS / user then a fallback (...) )

Comment (tiku-odoo) [reply content/administration/maintain/azure_oauth.rst:275]: Added this in, resolved issue.

File: content/administration/maintain/azure_oauth.rst [hunk 6]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
-of :guilabel:`notifications`. The fallback email must be configured as a :guilabel:`general
-transactional server`. The :guilabel:`mail.default.from` system parameter must be set to the
-:guilabel:`username` of the general transactional server account. For more information see
-:ref:`Use a default email address <email_communication/default>`.
+Configuring a single outgoing server is the simplest configuration available for Microsoft Azure
+and it doesn't require extensive access rights for the users in the database.
+
+A generic email address would be used to send emails for all users within the database. For example
+it could be structured with a `notifications` alias (`notifications@example.com`) or `contact` alias
+(contact@example.com). This address must be set as the :guilabel:`FROM Filtering` on the server.
```
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth.rst:201]: To avoid the example email formatting as a _mailto:_ link, escape the domain using `\` if this is the desired formatting output 🙂
```suggestion
(contact\@example.com). This address must be set as the :guilabel:`FROM Filtering` on the server.
```

File: content/administration/maintain/azure_oauth.rst [hunk 7]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+This address must also match the `{mail.default.from}@{mail.catchall.domain}` key combination in the
+system parameters.
+
+.. seealso::
+   For more information on the from filter visit :ref:`this documentation
+   <email_communication/default>`.
```
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth.rst:207]: I think this may be gray area, but I prefer to stay away from ambiguous link titles (similar to alt text on images). I would rather the link title to be descriptive. By removing the label we can just use the header the anchor is linking to, or provide a more descriptive label than "this documentation". What do you think? I will suggest this for all ambiguous link labels 😉
```suggestion
.. seealso::
   For more information on the from filter visit: :ref:`email_communication/default`.
```

File: content/administration/maintain/azure_oauth.rst [hunk 8]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
 
 .. note::
    The :guilabel:`System Parameters` can be accessed by activating
    :doc:`../../applications/general/developer_mode` in the :menuselection:`Settings --> Technical
    --> Parameters --> System Parameters` menu.
```
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth.rst:212]: Can use the :ref: here
```suggestion
   The :guilabel:`System Parameters` can be accessed by activating :ref:`developer-mode` in the
   :menuselection:`Settings --> Technical --> Parameters --> System Parameters` menu.
```

File: content/administration/maintain/azure_oauth.rst [hunk 9]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+
+.. image:: azure_oauth/from-name-remain.png
+   :align: center
+   :alt: Name from real sender with static email.
+
+.. example::
+   Single outgoing mail server configuration:
+
+   - Outgoing mail server **username** (login) = `notifications@example.com`
+   - Outgoing mail server  :guilabel:`FROM Filtering` = `notifications@example.com`
```
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth.rst:226]: Remove extra space
```suggestion
   - Outgoing mail server :guilabel:`FROM Filtering` = `notifications@example.com`
```

File: content/administration/maintain/azure_oauth.rst [hunk 10]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+Setup
+*****
+
+Each user should have a separate email server set up. The :guilabel:`FROM Filtering` should be set
+so that only the user's email is sent from that server. In other words, only a user with an email
+address that matches the set :guilabel:`FROM Filtering` is able to use this server.
+
+.. seealso::
+   For more information on the from filter visit :ref:`this documentation
+   <email_communication/default>`.
```
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth.rst:250]: Ambiguous link label
```suggestion
   For more information on the from filter visit: :ref:`email_communication/default`.
```

File: content/administration/maintain/azure_oauth.rst [hunk 11]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+
+A fallback server must be setup to allow for the sending of :guilabel:`notifications`. :ref:`See
+this documentation <azure_oauth/notifications>`. The :guilabel:`FROM Filtering` for this server
+should have the value of the `{mail.default.from}@{mail.catchall.domain}`.
```
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth.rst:254]: Ambiguous link label
```suggestion
A :ref:`fallback server <azure_oauth/notifications>` must be setup to allow for the sending of
:guilabel:`notifications`. The :guilabel:`FROM Filtering` for this server should have the value of
the `{mail.default.from}@{mail.catchall.domain}`.
```

File: content/administration/maintain/azure_oauth.rst [hunk 12]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+
+.. note::
+   The :guilabel:`System Parameters` can be accessed by activating
+   :doc:`../../applications/general/developer_mode` in the :menuselection:`Settings --> Technical
+   --> Parameters --> System Parameters` menu.
```
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth.rst:259]: Can use the :ref: here
```suggestion
   The :guilabel:`System Parameters` can be accessed by activating :ref:`developer-mode` in the
   :menuselection:`Settings --> Technical --> Parameters --> System Parameters` menu.
```

File: content/administration/maintain/azure_oauth.rst [hunk 13]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+
+.. important::
+   The configuration for this transactional email server can work alongside an outgoing mass-mailing
+   email server. The :guilabel:`FROM Filtering` for the mass-mailing email server can remain empty,
+   but it's require to be added in the Settings of the email marketing application.
```
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth.rst:264]: Capitalization and app name formatting
```suggestion
   but it's require to be added in the settings of the *Email Marketing* application.
```

File: content/administration/maintain/azure_oauth.rst [hunk 14]
```diff
@@ -188,24 +188,101 @@ configuration automatically loads the :guilabel:`token` in Odoo, and a tag stati
...
+
+   .. seealso::
+      For more information on setting the mass-mailing email server visit :ref:`this documentation
+      <email_communication/mass_mails>`.
```
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth.rst:268]: Ambiguous link label
```suggestion
      For more information on setting the mass-mailing email server visit:
      :ref:`email_communication/mass_mails`.
```

File: content/administration/maintain/azure_oauth/from-name-remain.png
Comment (samueljlieber) [inline content/administration/maintain/azure_oauth/from-name-remain.png:1]: This image doesn't appear to be compressed, please use PNGQuant 🙂

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @Abridbus and @tiku-odoo, this PR is looking good! I have just a couple technical changes and some suggestions, listed below. Please let me know if you have any questions and your thoughts on the link labels. 

Thank you for the continued hard work!!

File: content/administration/maintain/azure_oauth.rst (hunk 7 above)
Comment (StraubCreative) [reply content/administration/maintain/azure_oauth.rst:207]: I prefer this too.

**Comment by tiku-odoo** [comment]
@samueljlieber I've made the changes you requested. Thanks for your review of this doc. 

Ready to move to ZST, I believe. 

Thanks :+1:

**Review by samueljlieber — APPROVED** [review]
Looks good @tiku-odoo thanks! 🙂
On to you @StraubCreative

**Comment by StraubCreative** [comment]
LGTM 👍 
Thanks @Abridbus and team!

@robodoo r+
